Documentation/cubes.rst

```diff
@@ -33,11 +35,15 @@ A standard cube:
 A transparent cube is a cube without any forwarding capability, such as a network monitor, NAT and a firewall.
 
 A transparent cube:
-  - does not define any port, but has to be *attached* to an existing port such as the one of a normal service (e.g., port1 on router2) or a network interface in the host (e.g., veth0);
-  - cannot be *connected* to other services, but *attached*.
-  - it *inherits* all the parameters associated to ports (e.g., MAC/IPv4 addresses, link speed, etc.) from the actual port it is attached to (e.g., port 1 on router2, or veth0);
+  - does not define any port, hence it cannot be *connected* to other services, but *attached* to an existing port;
+  - it can be *attached* to an existing port such as the one of a normal service (e.g., port1 on router2) or a network device (netdev) in the host (e.g., `veth0` or `eth0`);
+  - it *inherits* all the parameters associated to ports (e.g., MAC/IPv4 addresses, link speed, etc.) from the actual port it is attached to;
   - it allows to have a stack of transparent services on top of a port, very similar to a stack of functions.
 
+A transparent cube can define two processing handlers, *ingress* and *egress*, which operate on two possible traffic directions:
+  - *ingress*: it handles the traffic that goes *toward* the port it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the TCP/IP stack and it is expected to *leave* the host from that port);
+  - *egress*: it handles the traffic that comes from the cube/netdev it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the external world and that is expected to go toward the TCP/IP stack of the host).
```

Comment:
```suggestion
  - *ingress*: it handles the traffic that goes *toward* the port it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the external world and that is expected to go toward the TCP/IP stack of the host);
  - *egress*: it handles the traffic that comes from the cube/netdev it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the TCP/IP stack and it is expected to *leave* the host from that port).
```